packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -63,6 +63,38 @@ class ErrorHandlingFileSystem extends ForwardingFileSystem {
     }
   }
 
+  /// Delete the file and return true if it exists, take no action and return
+  /// false if it does not.
+  ///
+  /// This method should be prefered to checking if it exists and
+  /// then deleting, because it handles the edge case where the file is
+  /// deleted by a different program between the two calls.
+  static bool deleteFileIfExists(File file, {bool recursive = false}) {
```

Comment:
How about:
```dart
static bool deleteIfExists(FileSystemEntity entity, ...) { ... }
```

Then it can handle directories and links as well.

Reply:
Will check if these have the same error code on windows

Reply:
same error code for directories